pilot/platform/cloudfoundry/servicediscovery.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudfoundry
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	copilotapi "code.cloudfoundry.org/copilot/api"
+
+	"istio.io/istio/pilot/model"
+)
+
+type CopilotClient interface {
+	copilotapi.IstioCopilotClient
+}
+
+const CF_APP_PORT = 8080
```

Comment:
nit: golang style for constants: `CF_APP_PORT` -> `CFAppPort`

Reply:
also, since we are already in the cloudfoundry package, maybe just `AppPort` to avoid stuttering?